### DISCUSSION TITLE
Expand/collapse subcollections

### DISCUSSION THREAD

**Discussion by cboulanger** [windingwind/zotero-actions-tags#357] [Ideas]
Does anyone know the API for expanding and collapsing the subcollections of a collection in the collection pane and can provide an action script that would allow to toggle between "all subcollections fully expanded" and "all subcollections fully collapsed"? Thank you.

**Comment by Ghiclgi** [comment]


**Reply by rdiaz02** [reply]
Thanks. It worked perfectly up to Zotero 8. With Zotero 8, it seems the setting is briefly toggled, to return (sometimes after several iterations) to the original value.

**Reply by mjthoraval** [reply]
As I understand, the solution proposed here is to toggle the "View" menu option "Show Items from Subcollections".
This is not what was asked originally by  @cboulanger?

**Reply by rdiaz02** [reply]
Yes, you are right. I think I was not clear. The problem is that the toggle is toggled an even number of times, so the state ends up being the original one. 

**Edit:** Well, I cannot reproduce it now. I am now on Zotero 8.0.2 (I was on 8.0.1 yesterday when I made the comment; maybe it affected only 8.0.1, or maybe I did something silly). Sorry for the noise!!

**Reply by iagogv3** [reply]
> This is not what was asked originally by @cboulanger?

Actually, he says " expanding and collapsing the subcollections of a collection **in the collection pane**". I understand the collection pane is the left-side pane, nor the middle one. I found this because I also was looking for this feature (not solved with the code), since, selected a collection, clicking `+`, expands all the collections in My Library, while I´d like to get expanded only the subcollections of the selected collection.

**Reply by rdiaz02** [reply]
Yes, I was referring (or trying to refer to) showing (or not) all items from subcollections.  For:

> (...) expanding and collapsing the subcollections of a collection **in the collection pane**".   (...)
> since, selected a collection, clicking `+`, expands all the collections in My Library, while I´d like to get expanded only the subcollections of the selected collection.

what I use are the arrow keys: right arrow key expands the subcollections immediately below the selected collection (but not recursively, just one level), and the left arrow key collapses subcollections.  (When no expansion/collapse of subcollections is possible, left and right arrows move my to the immediate parent or immediate child, respectively).

"+" and "-" expand/contract all subcollections (regardless of what collection I am on).

This works for me with Zotero 8.0.3 (and worked in previous 8s and I think all of 7).

**Reply by iagogv3** [reply]
Indeed, this is why my goal would be to expand recursively all the subcollections nested in the one selected.

**Comment by cboulanger** [comment]
As the original answer is no longer visible (or not for me at least), here is the code again, which works well for me (Zotero 8.0.2-beta.4+2f7055e15 - I cannot say anything about the non-beta versions).

```javascript
if(item) return;
ZoteroPane = Zotero.getActiveZoteroPane();
Zotero.Libraries.getAll().forEach(lib => 
    ZoteroPane.collectionsView.collapseLibrary(lib.libraryID))
````

**Reply by iagogv3** [reply]
Not working for me. It collapses always all the libraries, but do not  expand.

**Reply by Ghiclgi** [reply]
> As the original answer is no longer visible (or not for me at least), here is the code again, which works well for me (Zotero 8.0.2-beta.4+2f7055e15 - I cannot say anything about the non-beta versions).

I'm very sorry. I misunderstood your question so the wrong answer was deleted.

**Reply by mjthoraval** [reply]
@cboulanger I guess you are confusing with another action script that you shared here?
https://github.com/windingwind/zotero-actions-tags/discussions/406
The script you give does not seem to do what you requested here.

Thank you @Ghiclgi for moving your script to a separate page:
https://github.com/windingwind/zotero-actions-tags/discussions/558

**Comment by mjthoraval** [comment]
Here is a script written with the help of Claude.ai to expand all subcollections:
https://github.com/windingwind/zotero-actions-tags/discussions/559

**Reply by mjthoraval** [reply]
I have now added improved versions that are also working on item, for expanding (https://github.com/windingwind/zotero-actions-tags/discussions/562) and collapsing (https://github.com/windingwind/zotero-actions-tags/discussions/563).
